runtime/scheduler.cpp

```diff
@@ -73,13 +73,24 @@ Scheduler::Scheduler(const Scheduler::conf_t config) :
 }
 
 
-bool Scheduler::add_task(uint32_t repeat_period, Scheduler_task::run_mode_t run_mode, Scheduler_task::timing_mode_t timing_mode, Scheduler_task::priority_t priority, Scheduler_task::task_function_t call_function, Scheduler_task::task_argument_t function_argument, uint32_t task_id)
+bool Scheduler::add_task(uint32_t repeat_period,
+              Scheduler_task::task_function_t call_function,
+              Scheduler_task::task_argument_t function_argument,
+              Scheduler_task::priority_t priority,
+              Scheduler_task::timing_mode_t timing_mode,
+              Scheduler_task::run_mode_t run_mode,
+              int32_t task_id)
 {
     bool task_successfully_added = false;
 
     // Check if the scheduler is not full
     if (task_count_ < max_task_count_)
     {
+        if (task_id == -1)
```

Comment:
See commit  	ecb317b :  	*Modify param order when creating scheduler task, give default values*

The param task_id now has a default value of -1, in which case the ID is chosen automatically. Two reasons for this:
- you can create tasks in any order and at different locations without worrying about using twice the same ID
- the calls to `add_task` are shorter (most of the time)